### PR TITLE
[MHLO] Init MHLO slice like op patterns

### PR DESCRIPTION
**Pull request by fortianyou**
See RFC: https://github.com/llvm/torch-mlir/issues/999

Co-authored-by: Bairen Yi yibairen.byron@bytedance.com
Co-authored-by: Jiawei Wu xremold@gmail.com
Co-authored-by: Tianyou Guo tianyou.gty@alibaba-inc.com
Co-authored-by: Xu Yan yancey.yx@alibaba-inc.com
Co-authored-by: Ziheng Jiang ziheng.jiang@bytedance.com

### CODE REVIEW

**Comment by fortianyou** [comment]
This is a follow-up of PR: https://github.com/llvm/torch-mlir/pull/1090

Please review for me @silvasean @powderluv @ZihengJiang @byronyi @Yancey1989

**Review by ZihengJiang — APPROVED** [review]
LGTM. One minor suggestion, maybe we can merge those ops into ViewLikeOps or have a Uncategorized.cpp like TorchToLinalg

File: lib/Conversion/TorchToMhlo/PopulatePatterns.h [hunk 1]
```diff
@@ -19,6 +19,10 @@ namespace torch_to_mhlo {
 void populateBasicOpPatternsAndLegality(TypeConverter &typeConverter,
                                         RewritePatternSet &patterns,
                                         ConversionTarget &target);
+void populateSliceLikeOpPatternsAndLegality(TypeConverter &typeConverter,
+                                        RewritePatternSet &patterns,
```
Comment (silvasean) [inline lib/Conversion/TorchToMhlo/PopulatePatterns.h:23]: nit: format

**Review by silvasean — APPROVED** [review]

**Comment by silvasean** [comment]
I will let you folks iterate on the pattern details. LGTM overall.

**Comment by fortianyou** [comment]
> One minor suggestion, maybe we can merge those ops into ViewLikeOps or have a Uncategorized.cpp like TorchToLinalg

Thanks, I will merge these codes later.

**Pull request merged**